ftw/trash/utils.py

```diff
@@ -15,6 +15,17 @@ def is_trash_disabled():
     return os.environ.get('DISABLE_FTW_TRASH', None) == 'true'
 
 
+def is_migrating_plone_site(request):
+    """Detects whether we are in a request which migrats Plone to a new version.
+    In this case we want to disable the trash, because it may cause problems
+    e.g. when deleting tools.
+    """
+    return request.steps and request.steps[-1] in (
```

Comment:
What is request.steps doing? I'ts probably fine like that, I've just never heard of it.

Reply:
`request.steps` is a list of path segments of the request.
Example:
`http://localhost/foo/bar/folder_contents` will result in a `request.steps = ['foo', 'bar', 'folder_contents']`.

Reply:
Thanks!